### ISSUE TITLE
AuthorityKeyIdentifier property of the Extension object is NULL (missing) in the self signed certificate generated using node-opcua-crypto version 4.3.1

### ISSUE THREAD

**Issue by Graos** [node-opcua/node-opcua#1296]
node-opcua-crypto version: 4.3.1

As the title suggests AuthorityKeyIdentifier is NULL (missing) in the self signed certificate

ExploreCertificate function printed the following. As you can see there is no display of AuthorityKeyIdentifier after SubjectKeyIdentifier.

![image](https://github.com/node-opcua/node-opcua/assets/69800669/9b1fd581-089e-41ba-a20b-e8bb2d17abd8)

Code snippet as used from the Test code:

```
const { cert } = await createSelfSignedCertificate({
        privateKey,
        //notAfter: new Date(2023, 8, 5),
        //notBefore: new Date(2019, 1, 1),
        validity:30,
        dns: [osmodule.hostname()],
        subject: "/CN=NodeOPCUA-Server/O=ABC/L=XXX/ST=YYY/C=ZZ/DC=XCV.com",
        ip: epurls,
        applicationUri: applicationURI,
        purpose: CertificatePurpose.ForApplication,
    });
    const certificateDER = convertPEMtoDER(cert);
    const tmpCertificatePemFile = path.join(__dirname, "selfsignedCertificate.pem");
    await Fs.promises.writeFile(tmpCertificatePemFile, cert);
    const tmpCertificateDerFile = path.join(__dirname, "selfsignedCertificate.der");
    await Fs.promises.writeFile(tmpCertificateDerFile, certificateDER);
    const  privateKeyPem  = await privateKeyToPEM(privateKey);
    const tmpPrivateKeyFilename = path.join(__dirname, "privatekey.pem");
    await Fs.promises.writeFile(tmpPrivateKeyFilename, privateKeyPem.privPem);
    const info = exploreCertificate(convertPEMtoDER(cert));
    console.log(`${util.inspect(info)}`);

```

**Comment by Graos** [comment]
specification says:
<html>
<body>
<a name="Table36" style="box-sizing: border-box; color: rgb(33, 37, 41); text-decoration: none; font-family: system-ui, -apple-system, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, &quot;Liberation Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255);">

authorityKeyIdentifier | (no mapping) | Provides more information about the key used to sign the Certificate. It shall be specified for Certificates signed by a CA. It should be specified for self-signed Certificates.
-- | -- | --


</a>
</body>
</html>authorityKeyIdentifier

(no mapping)

Provides more information about the key used to sign the [Certificate](https://reference.opcfoundation.org/search/25?t=Certificate). It shall be specified for [Certificates](https://reference.opcfoundation.org/search/25?t=Certificates) signed by a CA. It should be specified for self-signed [Certificates](https://reference.opcfoundation.org/search/25?t=Certificates).

**Comment by erossignon** [comment]
Well spotted, thank you. I have copied your issue into the node-opcua-cryoto project.

**Comment by erossignon** [comment]
fix in [node-opcua@2.113.0](https://github.com/node-opcua/node-opcua/releases/tag/v2.113.0)
